query/src/storages/system/stages_table.rs

```diff
@@ -49,13 +49,38 @@ impl AsyncSystemTable for StagesTable {
         let mut copy_options: Vec<Vec<u8>> = Vec::with_capacity(stages.len());
         let mut file_format_options: Vec<Vec<u8>> = Vec::with_capacity(stages.len());
         let mut comment: Vec<Vec<u8>> = Vec::with_capacity(stages.len());
+        let mut number_of_files: Vec<Vec<u8>> = Vec::with_capacity(stages.len());
```

Comment:
It's hard to calculate the number of files for external stages, what value should the external stage use?

Reply:
`number_of_file`s should be optional, it is a `None/NULL` for external stages.